File: server/models/item.js
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var ItemSchema = new mongoose.Schema({
    authorId: {
    	type: String, 
    	required: true
    },
    alias: {
        type: String,
        required: true
    },
    dateBuy: {
        type: Date,
        required: true
    },
    dateEnd: {
        type: Date,
        required: true
    },
    dateNotification: {
        type: Date,
        required: true
    },
    dateCreation: {
        type: Date,
        required: true
    },
    duration: {
        type: String,
        required: true
    },
    emailNotification: {
        type: Boolean,
        required: true
    },
    smsNotification: {
        type: Boolean,
        required: true
    },
	images: {
		type: [String],
		required: false
	}
});


module.exports = mongoose.model('Item', ItemSchema);

File: AndroidApp/app/src/main/java/com/androidapp/models/Item.java
package com.androidapp.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Damien on 5/3/2017.
 */

public class Item {
    @SerializedName("_id")
    public String mId;

    @SerializedName("authorId")
    public String mAuthorId;

    @SerializedName("alias")
    public String mAlias;

    @SerializedName("dateBuy")
    public String mDateBuy;

    @SerializedName("dateEnd")
    public String mDateEnd;

    @SerializedName("dateNotification")
    public String mDateNotification;

    @SerializedName("dateCreation")
    public String mDateCreation;

    @SerializedName("duration")
    public String mDuration;

    @SerializedName("emailNotification")
    public Boolean mEmailNotification;

    @SerializedName("smsNotification")
    public Boolean mSmsNotification;

    @SerializedName("images")
    public List<String> mImages;

    public Item(String alias, String dateBuy, String dateEnd, String dateNotification,
                Boolean emailNotification, Boolean smsNotification) {
        this.mAlias = alias;
        this.mDateBuy = dateBuy;
        this.mDateEnd = dateEnd;
        this.mDateNotification = dateNotification;
        this.mEmailNotification = emailNotification;
        this.mSmsNotification = smsNotification;
    }
}
